packages/@aws-cdk/aws-stepfunctions-tasks/lib/emrcontainers/start-job-run.ts

```diff
@@ -118,21 +118,22 @@ export class EmrContainersStartJobRun extends sfn.TaskStateBase implements iam.I
     }
 
     if (props.jobDriver.sparkSubmitJobDriver?.entryPointArguments
-      && this.isArrayOfStrings(props.jobDriver.sparkSubmitJobDriver.entryPointArguments.value) === false
-      && !sfn.JsonPath.isEncodedJsonPath(props.jobDriver.sparkSubmitJobDriver.entryPointArguments.value)) {
-      throw new Error(`Entry point arguments must be a string array. Received ${props.jobDriver.sparkSubmitJobDriver.entryPointArguments.type}`);
+      && this.isArrayOfStrings(props.jobDriver.sparkSubmitJobDriver.entryPointArguments.value) === false) {
+      throw new Error(`Entry point arguments must be a string array. Received ${props.jobDriver.sparkSubmitJobDriver.entryPointArguments.type}.`);
     }
 
-    if (props.jobDriver.sparkSubmitJobDriver?.entryPointArguments
-      && (props.jobDriver.sparkSubmitJobDriver?.entryPointArguments.value.length > 10280
-        || props.jobDriver.sparkSubmitJobDriver?.entryPointArguments.value.length < 1)
-      && !sfn.JsonPath.isEncodedJsonPath(props.jobDriver.sparkSubmitJobDriver?.entryPointArguments.value)) {
-      throw new Error(`Entry point arguments must be an string array between 1 and 10280 in length. Received ${props.jobDriver.sparkSubmitJobDriver?.entryPointArguments.value.length}.`);
+    if (props.jobDriver.sparkSubmitJobDriver?.entryPointArguments) {
+      if ((typeof props.jobDriver.sparkSubmitJobDriver.entryPointArguments.type === 'string'
+        && !sfn.JsonPath.isEncodedJsonPath(props.jobDriver.sparkSubmitJobDriver.entryPointArguments.value))
+        || (props.jobDriver.sparkSubmitJobDriver.entryPointArguments.value.length > 10280
+          || props.jobDriver.sparkSubmitJobDriver.entryPointArguments.value.length < 1)) {
+        throw new Error(`Entry point arguments must be an string array between 1 and 10280 in length. Received ${props.jobDriver.sparkSubmitJobDriver?.entryPointArguments.value.length}.`);
+      }
     }
 
     if (props.jobDriver.sparkSubmitJobDriver?.sparkSubmitParameters
-      && (props.jobDriver.sparkSubmitJobDriver.sparkSubmitParameters.length > 102400
-        || props.jobDriver.sparkSubmitJobDriver.sparkSubmitParameters.length < 1)) {
+      && (props.jobDriver.sparkSubmitJobDriver?.sparkSubmitParameters.length > 102400
+        || props.jobDriver.sparkSubmitJobDriver?.sparkSubmitParameters.length < 1)) {
```

Comment:
the `?`s are actually unnecessary since Typescript should be able to infer `sparkSubmitJobDriver` is defined from the first branch of the conditional